merkledag/utils/diffenum.go

```diff
@@ -65,27 +65,34 @@ type diffpair struct {
 // getLinkDiff returns a changeset between nodes 'a' and 'b'. Currently does
 // not log deletions as our usecase doesnt call for this.
 func getLinkDiff(a, b node.Node) []diffpair {
-	have := make(map[string]*node.Link)
-	names := make(map[string]*node.Link)
+	ina := make(map[string]*node.Link)
+	inb := make(map[string]*node.Link)
+	var aonly []*cid.Cid
+	for _, l := range b.Links() {
+		inb[l.Cid.KeyString()] = l
+	}
 	for _, l := range a.Links() {
```

Comment:
It will be faster to just call `KeyString()` once, bind it to a local variable here. (also easier on memory)

Reply:
done